### PR TITLE
dbt-materialize: skip deploy_init checks if under CI and pushing to the same PR

### PR DESCRIPTION
**Pull request by sjwiesman**
### Motivation

By default, `deploy_init` will fail if the schema or cluster already exists and contains user defined objects. However, we don't want this check to run if someone is working on a PR and pushing to the same branch multiple times. Unforunately, github actions are not expressive enough to only run a command once. So instead, we tag each schema and cluster with a comment based on a CI environment variable on creation. If set, the checks will be skipped on subsequent invocations. 

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  -

### CODE REVIEW

File: misc/dbt-materialize/dbt/include/materialize/macros/utils/ci_utils.sql [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+{% endmacro %}
+
+{% macro check_schema_ci_tag(schema) %}
+    {% set ci_tag = env_var('CI_TAG', '') %}
+    {% if ci_tag == '' %}
+        {{ return(false) }}
+    {% endif %}
+
+    {% set query %}
+        SELECT comment = '{{ tag }}', comment
```
Comment (morsapaes) [inline misc/dbt-materialize/dbt/include/materialize/macros/utils/ci_utils.sql:72]: Lost in variables at this point, where is `tag` coming from? Adding some comments or a test would help navigate the code; it's hard (at least for me) to put the pieces of the workflow together on the fly.

File: misc/dbt-materialize/dbt/include/materialize/macros/utils/ci_utils.sql [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        FROM mz_internal.mz_comments
+        JOIN mz_clusters USING (id)
+        WHERE mz_clusters.name = lower(trim('{{ cluster }}'))
+    {% endset %}
+    {% set results = run_query(query) %}
+    {% if execute %}
+        {% if results|length > 0 %}
+            {% if results.rows[0][0] is false %}
+                {{ exceptions.raise_compiler_error("""
+                    Deployment schema """ ~ schema ~ """ already exists and is tagged
```
Comment (morsapaes) [inline misc/dbt-materialize/dbt/include/materialize/macros/utils/ci_utils.sql:53]: Are cluster and schema backwards in the compiler errors, or am I not getting what you're validating here?

File: misc/dbt-materialize/dbt/include/materialize/macros/deploy/deploy_promote.sql [hunk 3]
```diff
@@ -83,15 +74,6 @@
     {% if not cluster_exists(deploy_cluster) %}
         {{ exceptions.raise_compiler_error("Deployment cluster " ~ deploy_cluster ~ " does not exist") }}
     {% endif %}
-
-    {% if cluster_contains_sources_or_sinks(cluster) %}
```
[the remark this answers is not in the document]
Comment (sjwiesman) [reply misc/dbt-materialize/dbt/include/materialize/macros/deploy/deploy_promote.sql:87]: No, I accidentally called this check twice. It's still in there 😅

File: misc/dbt-materialize/dbt/include/materialize/macros/utils/ci_utils.sql (hunk 2 above)
Comment (sjwiesman) [reply misc/dbt-materialize/dbt/include/materialize/macros/utils/ci_utils.sql:53]: Oh yeah. I got that wildly backward. I'll add some tests.

**Comment by sjwiesman** [comment]
Sorry this was so gross. I've pushed a fix that includes tests.

File: misc/dbt-materialize/dbt/include/materialize/macros/utils/ci_utils.sql (hunk 2 above)
Comment (morsapaes) [reply misc/dbt-materialize/dbt/include/materialize/macros/utils/ci_utils.sql]: Oh, cool. I was questioning my sanity for a minute.

**Review by morsapaes — APPROVED** [review]

**Pull request merged**